File: DSAHW4/src/Edge.kt


class Edge : Comparable<Edge> {

    val node1 : Int
    val node2 : Int
    val weight : Int

    constructor(Node1 : Int = 0, Node2 : Int = 0, weight : Int = 0) {
        this.node1 = Node1
        this.node2 = Node2
        this.weight = weight
    }

    constructor(input : String) {
        val firstSpace = input.indexOf(" ")
        val secondSpace = input.indexOf(" ", firstSpace + 1)

        val node1 = input.substring(0, firstSpace).toInt()
        val node2 = input.substring(firstSpace + 1, secondSpace).toInt()
        val weight = input.substring(secondSpace+1).toInt()

        this.node1 = node1
        this.node2 = node2
        this.weight = weight
    }


    override fun compareTo(other: Edge): Int {

        return this.weight - other.weight
    }
}

File: DSAHW3CS/DSAHW3CS/Edge.cs
namespace DSAHW3CS;

public class Edge : Comparer<Edge>
{
    public int weight { get; private set; }
    public int node1 { get; private set; }
    public  int node2 { get; private set; }

    public Edge(int node1 = 0, int node2 = 0, int weight = 0)
    {
        this.weight = weight;
        this.node1 = node1;
        this.node2 = node2;
    }
    

    public Edge(string inputParam)
    {
        int firstSpace = inputParam.IndexOf(" ");
        int secondSpace = inputParam.IndexOf(" ", firstSpace + 1);

        int node1 = Convert.ToInt32(inputParam[..firstSpace]);
        int node2 = Convert.ToInt32(inputParam[++firstSpace..secondSpace]);
        int weight = Convert.ToInt32(inputParam[++secondSpace..]);

        this.weight = weight;
        this.node1 = node1;
        this.node2 = node2;
    }
    
    public override int Compare(Edge? x, Edge? y)
    {
        return x.weight.CompareTo(y.weight); //Just remember to make x.weight as the Edge's own weight
    }

}